packages/connect/src/router-http-client.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import type { ConnectRouter } from "./router.js";
+import type { UniversalClientFn } from "./protocol/universal.js";
+import type { UniversalHandler } from "./protocol/universal-handler.js";
+import { ConnectError } from "./connect-error.js";
+import { createAsyncIterable } from "./protocol/async-iterable.js";
+import { Code } from "./code.js";
+
+/**
+ * An in-memory {@link UniversalClientFn | http client} that can be used to route requests to a {@link ConnectRouter}
+ * by passing network calls. Useful for testing and calling in-process services.
+ *
+ * @param router The {@link ConnectRouter} to use.
+ * @returns The {@link UniversalClientFn} that can be passed to `createTransport` functions.
+ */
+export function createRouterHttpClient(
+  router: ConnectRouter
+): UniversalClientFn {
+  const handlerMap = new Map<string, UniversalHandler>();
+  for (const handler of router.handlers) {
+    handlerMap.set(handler.requestPath, handler);
+  }
+  return async (uClientReq) => {
+    const reqUrl = new URL(uClientReq.url);
+    const handler = handlerMap.get(reqUrl.pathname);
+    if (!handler) {
+      throw new ConnectError(
+        `RouterHttpClient: no handler registered for ${reqUrl.pathname}`,
+        Code.Unimplemented
+      );
+    }
+    const uServerRes = await handler({
+      body: uClientReq.body,
+      httpVersion: "1.1",
```

Comment:
For bidi-streaming, all clients will run into this code path on the handler: https://github.com/bufbuild/connect-es/blob/main/packages/connect/src/protocol/universal-handler.ts#L244-L247

Let's pass `2.0` here. We have to keep in mind that we might want to have this configurable at some point.